### DISCUSSION TITLE
GeometryFeature rendering performance

### DISCUSSION THREAD

**Discussion by IldarKhayrutdinov** [Mapsui/Mapsui#1686] [General]
My Windows desktop app draws a lot of GeometryFeature objects (I'm use Mapsui 4.0-beta5).
Do I understand correctly that the rendering performance of these objects will match on all platforms: Wpf, WinUI, Avalonia?

**Comment by charlenni** [comment]
I assume not. Could only be, if all Skia implemententions have the same overhead. But they should have on the same hardware nearly the same.

**Reply by inforithmics** [reply]
For example the Maui.Winui Target is currently not GPU accelerated so Mapsui in MAUI on Windows is slower. But the other targets should be GPU acceleratoed.

**Reply by inforithmics** [reply]
But the to be sure performance wise measure it in a profiler. Because Maui runs on .NET 6 and .NET 6 is sometimes 50% faster than for example the XamarinForms runtime. But Best to measure when you evaluate.

**Reply by IldarKhayrutdinov** [reply]
thanks, oh yes, I meant on the same hardware
Windows 10 or 11
.Net 6 or .Net 7 preview

**Reply by wanglehui** [reply]
By read the code class Mapsui.UI.Maui.MapControl, GPU does not work currently on MAUI
(https://github.com/mono/SkiaSharp/issues/1893)

And we test the performance between MAUI for Android and Mapsui.UI.Android, the No-GPU  version performance  is obvious poor.
